### PR TITLE
Fix errors for all dancing-dots submissions

### PR DESCRIPTION
**Pull request by angelikatyborska**
Resolves https://github.com/exercism/elixir-representer/issues/96

In a separate PR, I would love to add some sort of smoke test that would run the formatter for every real exercise and check that it produced a representation. It would require to correctly include the exemplar solution + extra files ( just the exemplar doesn't cause a crash for dancing-dots), or maybe just run it on an empty solution? Not sure.

### CODE REVIEW

File: lib/representer.ex [hunk 1]
```diff
@@ -197,7 +197,8 @@ defmodule Representer do
     {{:., meta, path}, meta2, args}
   end
 
-  defp remove_type_parentheses({type, meta, args}) when args == [] or is_atom(args) do
+  defp remove_type_parentheses({type, meta, args})
+       when type != :%{} and (args == [] or is_atom(args)) do
```
Comment (angelikatyborska) [inline lib/representer.ex:201]: The issue was caused by us replacing `{%{}, [], []}` with `{%{}, [], nil}`, which apparently is not valid Elixir AST, or at least the formatter doesn't understand it.

Comment (jiegillet) [reply lib/representer.ex:201]: I suppose `{%{}, [], nil}` doesn't really make sense? 🤷 
Nice find

**Review by jiegillet — APPROVED** [review]
I suppose it

**Comment by jiegillet** [comment]
> In a separate PR, I would love to add some sort of smoke test that would run the formatter for every real exercise and check that it produced a representation. It would require to correctly include the exemplar solution + extra files ( just the exemplar doesn't cause a crash for dancing-dots), or maybe just run it on an empty solution? Not sure.

We could do that in the CI fairly easily. I did something like this in the Elm Analyzer, it pulls the Elm repo and check all exercises.

**Comment by angelikatyborska** [comment]
@ErikSchierboom can I or you do something to re-run the analyzer on all dancing-dots solutions for Elixir?

**Comment by angelikatyborska** [comment]
Hmm, or maybe this happens automatically

**Comment by ErikSchierboom** [comment]
@angelikatyborska I was already enqueuing the submissions for reprocessing :) But yes, it can take a while

**Comment by ErikSchierboom** [comment]
It's working :) https://exercism.org/tracks/elixir/exercises/dancing-dots/solutions

**Pull request merged**